app/controllers/concerns/foreman/controller/registration_command.rb

```diff
@@ -0,0 +1,66 @@
+module Foreman::Controller::RegistrationCommand
```

Comment:
Could this be a separate class, that would get the arguments and build the command?
It just seems as bit too much responsibility for the controller.

Btw such class could most definitelly include `ActiveModel` and act as a resource class (for forms and controllers).

Reply:
Created redmine: https://projects.theforeman.org/issues/31849